src/CodeGen_PTX_Dev.cpp

```diff
@@ -791,6 +809,256 @@ bool CodeGen_PTX_Dev::supports_atomic_add(const Type &t) const {
 
 }  // namespace
 
+class IsMatrixMultiply : public IRVisitor {
+    using IRVisitor::visit;
+
+public:
+    bool matmul_found = false;
+
+    Expr A;
+    Expr B;
+    Expr C;
+
+    std::string k_var_name;
+
+    IsMatrixMultiply(std::string k_var)
+        : k_var_name{std::move(k_var)} {
+    }
+
+    void visit(const Store *store) override {
+        //  matmul$1[t26] = matmul$1[t26] + float32((A[((A.stride.1*t34) - t33) + matmul$1.s1.k$x]*B[(B.stride.1*matmul$1.s1.k$x) + t35]))
+        const Expr wild_f32x = Variable::make(Float(32), "*");
+        vector<Expr> matches;
+        const Expr acc_pattern = wild_f32x + wild_f32x;
+        if (expr_match(acc_pattern, store->value, matches)) {
+            // matmul[t26]
+            const Load *matmul_load = matches[0].as<Load>();
+
+            // float32((A[((A.stride.1*t34) - t33) + matmul$1.s1.k$x]*B[(B.stride.1*matmul$1.s1.k$x) + t35]))
+            const Cast *cast = matches[1].as<Cast>();
+
+            if (matmul_load && cast) {
+                // Check if the load is loading from the same place where the store is storing
+                // i.e. if this is an update operation
+                if (matmul_load->name == store->name && equal(matmul_load->index, store->index)) {
+                    // Yes, this is an update operation, now let's check cast to see if it is a matmul that can be converted
+                    // to wmma intrinsic
+
+                    // float32((A[((A.stride.1*t34) - t33) + matmul$1.s1.k$x]*B[(B.stride.1*matmul$1.s1.k$x) + t35]))
+                    const Expr wild_f16x = Variable::make(Float(16), "*");
+                    const Expr mul_pattern = wild_f16x * wild_f16x;
+                    if (expr_match(mul_pattern, cast->value, matches)) {
+                        // A[((A.stride.1*t34) - t33) + matmul$1.s1.k$x]
+                        const Load *load_a = matches[0].as<Load>();
+                        // B[(B.stride.1 * matmul$1.s1.k$x) + t35]
+                        const Load *load_b = matches[1].as<Load>();
+
+                        if (load_a && load_b) {
+                            const Expr wild_i32x = Variable::make(Int(32), "*");
+                            const Expr k_var_expr = Variable::make(Int(32), k_var_name);
```

Comment:
This appears to be unused, I think it should be part of the patterns?

Reply:
Ah, I left it by mistake. Please see my comment https://github.com/halide/Halide/pull/5995#discussion_r631082738

Reply:
Removed in dae5a5812b42c1e573e439ed1266b9a156d232d6